### PR TITLE
Router

### PR DESCRIPTION
**Pull request by sri2k1us**
- Break up large router file according to end-points.
- Note this is on top of #130. Review from https://github.com/cyverse-de/sonora/pull/131/commits/0f2f254b7cd601f341ba2539d06f804e58b3ea79

### CODE REVIEW

File: src/server/api/user.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    api.get(
+        "/user-info",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "GET",
+            pathname: "/secured/user-info",
+        })
+    );
+
+    logger.info("adding the GET /api/dashboard handler");
```
Comment (ash3rz) [inline src/server/api/user.js:24]: I wouldn't think to look here in the `user` file to find this endpoint.  Might just be me though.

File: src/server/api/data.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        terrainHandler({
+            method: "POST",
+            pathname: "/secured/filetypes/type",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info("adding the POST /share handler");
```
Comment (ash3rz) [inline src/server/api/data.js:130]: Is this endpoint exclusive to data sharing?  I think we use this for apps and analyses sharing as well, which might make this hard to find in the `data` file

File: src/server/api/user.js (hunk 1 above)
Comment (sri2k1us) [reply src/server/api/user.js:24]: I was debating between putting it in a separate file vs putting it `user` and ultimately decided to put it here. I will move it a separate file.

File: src/server/api/data.js (hunk 2 above)
Comment (sri2k1us) [reply src/server/api/data.js:130]: Thats true. I will move this a separate file.

File: src/server/index.js [hunk 3]
```diff
@@ -68,8 +81,20 @@ app.prepare()
             getNotifications(authn.getUserID(request), ws);
         });
 
-        logger.info("adding the api router to the express server");
-        server.use("/api", apiRouter());
+        logger.info(
+            "$$$$$$$$ adding the api router to the express server $$$$$$$$$"
```
Comment (psarando) [inline src/server/index.js:85]: 🤑

**Comment by sri2k1us** [comment]
Thanks for the reviews.

File: src/server/api/router.js [hunk 4]
```diff
@@ -1,551 +0,0 @@
...
-        "/quicklaunches",
-        auth.authnTokenMiddleware,
-        terrainHandler({
-            method: "POST",
-            pathname: "/quicklaunches",
-            headers: {
-                "Content-Type": "application/json",
-            },
-        })
-    );
```
Comment (psarando) [inline src/server/api/router.js:367]: Oops, missed this in my review, but it looks like this endpoint wasn't ported into a new file, so it's missing now.

Comment (sri2k1us) [reply src/server/api/router.js:367]: 😖

**Pull request merged**